### ISSUE TITLE
dashboard CrashLoopBackOff: x509: certificate is valid for 192.168.99.100, 10.0.0.1, not 10.96.0.1

### ISSUE THREAD

**Issue by ddnomad** [kubernetes/minikube#2623]
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
BUG REPORT

Please provide the following details:

**Environment**:

**Minikube version**: v0.23.0
- **OS**: Arch Linux
- **VM Driver**: virtualbox
- **ISO version**: minikube-v0.23.6.iso
- **Install tools**: `pacaur -S minikube-git`
- **Others**: `pacaur -S minikube` fails as well

**What happened**:
```
$ minikube start
...
$ kubectl get pods --all-namespaces
NAMESPACE     NAME                          READY     STATUS             RESTARTS   AGE
kube-system   kube-addon-manager-minikube   1/1       Running            0          8m
kube-system   kube-dns-86f6f55dd5-ztl9k     2/3       Running            9          8m
kube-system   kubernetes-dashboard-mqp9n    0/1       CrashLoopBackOff   6          8m
$ kubectl logs kubernetes-dashboard-mqp9n --namespace=kube-system
2018/03/21 12:18:01 Starting overwatch
2018/03/21 12:18:01 Using in-cluster config to connect to apiserver
2018/03/21 12:18:01 Using service account token for csrf signing
2018/03/21 12:18:01 No request provided. Skipping authorization
2018/03/21 12:18:01 Error while initializing connection to Kubernetes apiserver. This most likely means that the cluster is misconfigured (e.g., it has invalid apiserver certificates or service accounts configuration) or the --apiserver-host param points to a server that does not exist. Reason: Get https://10.96.0.1:443/version: x509: certificate is valid for 192.168.99.100, 10.0.0.1, not 10.96.0.1
Refer to the troubleshooting guide for more information: https://github.com/kubernetes/dashboard/blob/master/docs/user-guide/troubleshooting.md
```

**What you expected to happen**:
No `CrashLoopBackOff` of K8S dashboard. I even have not restarted `minikube`. It's a fresh installation.

**How to reproduce it** (as minimally and precisely as possible):
Install `minikube-git` from Arch User Repository (AUR) and run the commands above.

**Output of `minikube logs` (if applicable)**:
See attachment: [logs.txt](https://github.com/kubernetes/minikube/files/1833185/logs.txt)

**Anything else do we need to know**:
I've tried deleting secretes with no success whatsoever. I get a reason why it happens (wrong certs) though I'm totally not sure it's an expected behaviour from a fresh installation.

**Comment by ddnomad** [comment]
Ok, installed `v25.0` and it works (at least without restart). I'll leave it here meanwhile if somebody would encounter the same issue.

**Comment by abh** [comment]
The workaround mentioned in #2765 to use `minikube start --bootstrapper=localkube` solved similar symptoms (on 0.28.0) for me.

**Comment by tstromberg** [comment]
This should be obsolete at this point. The dashboard setup has been refactored twice since this was reported. Please re-open if it's still an issue.
